06-building-a-program-start-to-finish/writing-tests/output/inverted-index.spec.js

```diff
@@ -0,0 +1,184 @@
+      .toEqual(undefined);
+    expect(invertedIndex.getIndex('books4'))
+      .toEqual(undefined);
+  });
+  it(`should verify that the index of JSON file can be gotten 
+  after file indexed`, () => {
+    expect(invertedIndex.getIndex('books.json'))
+      .toBeDefined();
+  });
+  it(`should map the string keys to the correct objects in 
+  the JSON array`, () => {
+    expect(invertedIndex.getIndex('adventure-books.json'))
+      .toEqual(
+        {
+          king: [0],
+          solomons: [0],
+          mines: [0],
+          it: [0],
+          tells: [0],
+          of: [0],
+          a: [0, 1],
+          search: [0],
+          an: [0],
+          unexplored: [0],
+          region: [0],
+          africa: [0],
+          by: [0],
+          group: [0],
+          adventurers: [0],
+          treasure: [1],
+          island: [1],
+          wild: [1],
+          seaman: [1],
+          billy: [1],
+          bones: [1],
+          comes: [1],
+          to: [1],
+          stay: [1],
+          bringing: [1],
+          with: [1],
+          him: [1],
+          large: [1],
+          sea: [1],
+          chest: [1]
+        }
+      );
+  });
+});
+
+describe('Search index functionality', () => {
+  it(`should return an array of the indices of the correct objects that contain
+    the words in the search query`, () => {
+    expect(invertedIndex
+      .searchIndex('a rabbit alliance with man', 'books.json'))
+      .toEqual({
+        'books.json': {
+          a: [0, 1],
+          rabbit: [0],
+          alliance: [1],
+          with: [],
+          man: [1]
+        }
+      });
+    expect(invertedIndex.searchIndex('a king billy', 'adventure-books.json'))
+      .toEqual({
+        'adventure-books.json': {
+          a: [0, 1],
+          king: [0],
+          billy: [1]
+        }
+      });
+  });
+  it('should normalize search string before search', () => {
+    expect(invertedIndex
+      .searchIndex('A RABBIT!@#$%^&*()+=-][}{/?><.,|}]`~€‹›', 'books.json'))
+      .toEqual({
+        'books.json': {
+          a: [0, 1],
+          rabbit: [0]
+        }
+      });
+  });
+  it(`should return an object with empty indices books key if no
+    match is found`, () => {
+    expect(invertedIndex.searchIndex('house', 'books.json')).toEqual({
+      'books.json': {
+        house: []
+      }
+    });
+  });
+  it(`should search all indexed files if second argument not passed to
+    searchIndex function`, () => {
+    expect(invertedIndex.searchIndex('alice king ring tells about a hobbit'))
+      .toEqual({
+        'adventure-books.json': {
+          alice: [],
+          king: [0],
+          ring: [],
+          tells: [0],
+          about: [],
+          a: [0, 1],
+          hobbit: []
+        },
+        'books.json': {
+          alice: [0],
+          king: [],
+          ring: [1],
+          tells: [],
+          about: [],
+          a: [0, 1],
+          hobbit: [1]
+        }
+      });
+  });
+});
```

Comment:
Kindly address the few issues and many style violations. Other thank that! Great work!